src/main/java/com/puppycrawl/tools/checkstyle/checks/modifier/InterfaceMemberImpliedModifierCheck.java

```diff
@@ -0,0 +1,423 @@
+     * on interface nested types.
+     * @param violateImpliedPublicNested
+     *        True to perform the check, false to turn the check off.
+     */
+    public void setViolateImpliedPublicNested(boolean violateImpliedPublicNested) {
+        this.violateImpliedPublicNested = violateImpliedPublicNested;
+    }
+
+    /**
+     * Setter to control whether to enforce that {@code static} is explicitly coded
+     * on interface nested types.
+     * @param violateImpliedStaticNested
+     *        True to perform the check, false to turn the check off.
+     */
+    public void setViolateImpliedStaticNested(boolean violateImpliedStaticNested) {
+        this.violateImpliedStaticNested = violateImpliedStaticNested;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.METHOD_DEF,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.CLASS_DEF,
+            TokenTypes.ENUM_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ScopeUtil.isInInterfaceBlock(ast) && !ScopeUtil.isInCodeBlock(ast)) {
+            switch (ast.getType()) {
+                case TokenTypes.METHOD_DEF:
+                    processMethod(ast);
+                    break;
+                case TokenTypes.VARIABLE_DEF:
+                    processField(ast);
+                    break;
+                case TokenTypes.CLASS_DEF:
+                case TokenTypes.INTERFACE_DEF:
+                case TokenTypes.ENUM_DEF:
+                    processNestedType(ast);
+                    break;
+                default:
+                    throw new IllegalStateException(ast.toString());
+            }
+        }
+    }
+
+    /**
+     * Check method on interface.
+     * @param ast method AST
+     */
+    private void processMethod(DetailAST ast) {
+        final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
+        if (violateImpliedPublicMethod
+                && modifiers.findFirstToken(TokenTypes.LITERAL_PRIVATE) == null
+                && modifiers.findFirstToken(TokenTypes.LITERAL_PUBLIC) == null) {
+            log(ast, MSG_KEY, PUBLIC_ACCESS_MODIFIER);
+        }
+        if (violateImpliedAbstractMethod
+                && modifiers.findFirstToken(TokenTypes.LITERAL_PRIVATE) == null
+                && modifiers.findFirstToken(TokenTypes.LITERAL_STATIC) == null
+                && modifiers.findFirstToken(TokenTypes.LITERAL_DEFAULT) == null
+                && modifiers.findFirstToken(TokenTypes.ABSTRACT) == null) {
+            log(ast, MSG_KEY, ABSTRACT_KEYWORD);
+        }
+    }
+
+    /**
+     * Check field on interface.
```

Comment:
`Check field on interface` => `Checks field in interface`

Reply:
Done